js/components/src/streamplace-store/streamplace-store.tsx

```diff
@@ -40,6 +46,46 @@ export const makeStreamplaceStore = ({
 }: {
   url: string;
 }): StoreApi<StreamplaceState> => {
+  // get local storage if exists
+  const getStorage = () => {
+    if (typeof window !== "undefined" && window.localStorage) {
+      return window.localStorage;
+    }
+    return null;
+  };
+
+  const storage = getStorage();
+  const VOLUME_STORAGE_KEY = "globalVolume";
+  const MUTED_STORAGE_KEY = "globalMuted";
+
+  // Load initial volume state from storage
+  let initialVolume = 1.0;
+  let initialMuted = false;
+
+  if (storage) {
+    try {
+      const storedVolume = storage.getItem(VOLUME_STORAGE_KEY);
+      const storedMuted = storage.getItem(MUTED_STORAGE_KEY);
+
+      if (storedVolume) {
+        const parsedVolume = parseFloat(storedVolume);
+        if (
+          Number.isFinite(parsedVolume) &&
+          parsedVolume >= 0 &&
+          parsedVolume <= 1
+        ) {
+          initialVolume = parsedVolume;
+        }
+      }
+
+      if (storedMuted) {
+        initialMuted = storedMuted === "true";
+      }
+    } catch (e) {
+      console.warn("Failed to load volume settings from storage:", e);
+    }
+  }
+
```

Comment:
Might be time to port over the Storage class into `components`, yeah? That'll give us this and also work on native. https://github.com/streamplace/streamplace/tree/bdf804a95970207319c60a303ddb9ad1d7800a8d/js/app/storage